java-client/src/main/java/co/elastic/clients/transport/rest_client/RestClientHttpClient.java

```diff
@@ -241,6 +251,51 @@ public void close() throws IOException {
         }
     }
 
+    public class RepeatableBodyResponse extends RestResponse {
+
+        BinaryData repeatableBody;
+
+        RepeatableBodyResponse(org.elasticsearch.client.Response restResponse) {
+            super(restResponse);
+        }
+
+        @Nullable
+        @Override
+        public BinaryData body() throws IOException {
+            if(repeatableBody != null) {
+                return repeatableBody;
+            }
+            BinaryData body = super.body();
+            if (body != null) {
+                if(body.isRepeatable()){
+                    repeatableBody = body;
+                }
+                else{
+                    repeatableBody = new ByteArrayBinaryData(body);
+                }
+            }
+            return repeatableBody;
+        }
+
+        public String getOriginalBodyAsString() throws IOException {
```

Comment:
Looks like this is used only in tests. Let's not add in the public API some things that aren't required to use the library.

Reply:
I actually wanted to provide this as a utility method to let users get the string directly